plugins/allocators/bitmap/bitmap_ipv4.go

```diff
@@ -69,7 +69,7 @@ func (a *IPv4Allocator) Allocate(hint net.IPNet) (n net.IPNet, err error) {
 
 	var next uint
 	// First try the exact match
-	if a.bitmap.Test(hintOffset) {
+	if !a.bitmap.Test(hintOffset) {
```

Comment:
and it'll always test this as well if the hint is invalid. I'll send a follow-up patch to test the hint only if it's valid